src/crap.nim

```diff
@@ -4,37 +4,35 @@ import
   strutils,
   times
 
-proc crap*(path: var string) =
-  var trashHome: string
+proc crap*(path: string) =
   let fullPath = expandFilename(path)
-  path = extractFilename(fullPath)
+  let fname = extractFilename(fullPath)
 
-  if existsEnv("XDG_DATA_HOME"):
-    trashHome = joinPath(getEnv("XDG_DATA_HOME"), "Trash")
+  let trashHome = if existsEnv("XDG_DATA_HOME"):
+    getEnv("XDG_DATA_HOME") / "Trash"
   else:
-    trashHome = joinPath(getHomeDir(), ".local/share/Trash")
+    getHomeDir() / ".local/share/Trash"
 
-  moveFile(fullPath, joinPath(trashHome, "files", path))
+  moveFile(fullPath, trashHome / "files" / fname)
 
   let 
     t = getTime()
-    formattedTime = t.format("yyyy-MM-dd") & "T" & t.format("HH:MM:ss")
+    formattedTime = t.format("yyyy-MM-dd'T'HH:MM:ss")
```

Comment:
This is documented in official docs for the [times](https://nim-lang.org/docs/times.html) module if you're wondering:
```
Other strings can be inserted by putting them in ''. 
```